src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs

```diff
@@ -338,6 +338,20 @@ protected virtual unsafe IntPtr AppWndProc(IntPtr hWnd, uint msg, IntPtr wParam,
 
                         break;
                     }
+                case WindowsMessage.WM_CONTEXTMENU:
+                    {
+                        var posPix = PointFromLParam(lParam);
+                        if (posPix.X == -1 && posPix.Y == -1)//this pos means it launched by keyboard
+                        {
+                            //keyboard shortcuts are resolved inside avalonia via handling of keyUP events...
+                            break;
+                        }
+
+                        var pos = DipFromLParam(lParam);
+                        KeyboardDevice.Instance.FocusedElement?.RaiseEvent(new ContextRequestedEventArgs(pos));
```

Comment:
Not sure that the `WindowImpl` should be reaching into the `KeyboardDevice` to get the currently focused element here.  The pattern used elsewhere is that events from the platform impl are delegated to the owner `Window` using `RawInputEventArgs`.